primes/ocaml/compute.ml

```diff
@@ -9,22 +9,28 @@ let is_prime ~primes ~prime_count ~candidate =
   in
   loop ~index:0
 
-let compute_primes ~max_prime_count =
+let compute_primes ~max_prime_count ~bench_debug =
   let primes = Array.make max_prime_count 0 in
   let rec loop ~prime_count ~candidate =
     if prime_count = max_prime_count then
       ()
     else if is_prime ~primes ~prime_count ~candidate then (
       primes.(prime_count) <- candidate;
-      Printf.printf "%d\n" candidate;
       loop ~prime_count:(prime_count + 1) ~candidate:(candidate + 1)
     ) else
       loop ~prime_count ~candidate:(candidate + 1)
   in
-  loop ~prime_count:0 ~candidate:2
+  loop ~prime_count:0 ~candidate:2;
+  if bench_debug == true then
+    let print prime =
+      Printf.printf "%d\n" prime in
+    Array.iter print primes
```

Comment:
It's easy to abuse the feature but here you could write the following without making the code too cryptic:

 ```ocaml
if bench_debug then
    Array.iter (Printf.printf "%d\n") primes
```

Reply:
Done